src/helpers/cycled.js

```diff
@@ -0,0 +1,63 @@
+import {Subject} from 'rx'
+import {
+  last, init, compose, map, fromPairs, intersection, keys,// omit,
+} from 'ramda'
+
+/**
+ * Cycle component sinks back to the component sources.
+ *
+ * Imagine you have a button component that takes a disabled$ stream:
+ *
+ *   Button({...sources, disabled$: ?})
+ *
+ * You want to disable the button when the button is clicked:
+ *
+ *   const button = Button({...sources, disabled$: button.click$})
+ *
+ * But this doesn't work, you don't have the click$ source yet. This function
+ * let's you cycle that click$ source back to the button:
+ *
+ *   const DisableOnClickButton = cycled('click$', sources =>
+ *     Button({
+ *       ...sources,
+ *       disabled$: sources.click$.map(true).startWith(false)
+ *     })
+ *   )
+ *
+ *   DisableOnClickButton({...sources})
+ *
+ * @param streamNames array of stream names
+ * @param component
+ * @returns {function(*names, component)}
+ */
+export function cycled(...streamNames) {
```

Comment:
I'm almost 100% certain that this will cause memory leaks. A possible fix is here to that https://gist.github.com/TylorS/5c514b05b45620d6164d76af82e2fa7d